utils/build-script

```diff
@@ -53,6 +53,15 @@ from swift_build_support.swift_build_support.toolchain import host_toolchain
 # TODO: Remove this constant, it's really not helpful.
 HOME = os.environ.get("HOME", "/")
 
+_SUPPORTED_XCODE_BUILDS = [
+    ("12.0 beta 3", "12A8169g"),
+    ("12.0 beta 4", "12A8179i"),
+    ("12.0 beta 5", "12A8189h"),
+    ("12.0 beta 6", "12A8189n"),
+    ("12.0 GM 1", "12A7208"),
+    ("12.0 GM 2", "12A7209"),
+    ("12.0.1", "12A7300"),
```

Comment:
Also these current values are an assumption, the README says 12.0b3, and therefore I assume up to 12.0.1 works

Reply:
I am little worried about having list of supported Xcode because it will be come stale overtime. Should we only provide what is support in CI?

Reply:
Yea that's a good question. I like the idea of being permissive here especially beta to beta because it's such a high cost to download a new one / unxip it if it's still supported "enough." And ideally this wouldn't get super stale because at the point support was added / removed for specific Xcode versions it would be updated. But I can see that being tough especially if apple folks are all skipping this check.

Reply:
This would be still tested on open source CI, so it will not go stale if we pick one Xcode version.

Reply:
Some thoughts:

* It seems like it'll be easier to maintain a single version, instead of a list
* Having a list seems like it'll expand the chances of one or more being incorrect, since there will be no CI to validate the list
* Ideally the single version would match the version used on CI, but at this moment in time that's a beta which most people have likely moved away from
* Having a long list is an optimization, maybe it's better to be correct first

What if we limit the list to one or two versions. For example right now we could say either 12b3 (CI), or 12.0.1 (a release). When CI and latest version match, then there'd be only one version.

Reply:
Definitely, yea if you think that's the best way to do it we can do 1, but I do think if possible we should allow as many as should actually work for flexibility, especially during times where there are so many Xcode releases

Reply:
I'm thinking we can put a comment on `_SUPPORTED_XCODE_BUILDS` and say something like:

> These versions are community sourced. At any given time only the Xcode version used by Swift CI is officially supported. See https://ci.swift.org.

Maybe we could even separate out the current CI version of Xcode into its own constant, and if the current version doesn't match that, a warning/info notice can be shown, to highlight the potential issue in case anything goes wrong.

Reply:
I like this idea, pushed this comment. I don't immediately see anything in the repo tying us to that version, so I'm not sure we can easily setup the second constant (it is in the README, but I don't see it elsewhere)

Reply:
I figured the second constant would have to be defined here. The question is: Is it good to introduce that constant here, or not?